### DISCUSSION TITLE
The Future of Tephi

### DISCUSSION THREAD

**Discussion by ESadek-MO** [SciTools/tephi#162] [Ideas]
<p align = "center">
    <img src="https://github.com/user-attachments/assets/09181885-da9c-4be8-87c0-ccd0f9e7c213" hspace="30" >
    <img src="https://github.com/user-attachments/assets/09181885-da9c-4be8-87c0-ccd0f9e7c213" hspace="30" >
    <img src="https://github.com/user-attachments/assets/09181885-da9c-4be8-87c0-ccd0f9e7c213" hspace="30" >
</p>

[Project Board](https://github.com/orgs/SciTools/projects/39/views/1)

Tephi is getting some much needed TLC, and we want to hear from you! What kinda things would you like to see implemented in the future? What features would you like to see, what bugs would you like fixed, what documentation would you like updating?

Feel free to leave a comment here, or if you'd rather:
Want to request a change? Create an [issue](https://github.com/SciTools/tephi/issues/new/choose)!
Want to add a change yourself? Create a [pull request](https://github.com/SciTools/tephi/compare)!

**Comment by hilaryweller0** [comment]
There is documentation that includes the class https://notebook.community/SciTools/tephi/index
tephi.Tephigram
that seems to have  useful additional functionality. Do you know of any plans to make it available?

**Reply by ESadek-MO** [reply]
Hi Hilary, this was added in the latest release (0.4.0.post0, linked [here](https://github.com/SciTools/tephi/releases/tag/v0.4.0.post0)). Note that this was somewhat of a rehaul, so included a number of breaking changes.

Please do try it out! It's fairly new, and I'd really appreciate comments on both the changes and the documentation!

**Reply by hilaryweller0** [reply]
I cannot find a class or function called Tephigram in 0.4.0.post0 or 0.4.0.dev.
But I am happy to use TephiAxes. 
I am trying to plot just the bottom part of a tephigram, not going above about 500 hPa. I cannot work out how to do that. Here is a mwe:

```python
import matplotlib.pyplot as plt
import tephi

# Data goes up to 200 hPa
P = [1000, 750, 500, 200]
T = [20, 0, -10, -60]

# I want the tephigram to stop at 500 hPa
tpg = tephi.TephiAxes()
tpg.set(ylim=(1000,500))
tpg.add_isobars(min_theta=-40)
tpg.plot(zip(P,T))
plt.show()
```

Do I need to truncate the data in order to truncate the tephigram?
thanks
Hilary

**Reply by trexfeathers** [reply]
Hi @hilaryweller0,

Sorry, we've confused matters on a few fronts! Here are some clarifications:

1. `TephiAxes` is the new name for `Tephigram` (not the other way round as we previously implied). It should behave mostly the same as it used to, although it has been heavily reworked to give Tephi a brighter future.
2. The notebook you reference was missed when Tephi was updated to version 0.4. It demonstrates correct usage of Tephi version 0.3. You are still free to install version 0.3, or you can take a look at the updated notebook I created today (in #253, pending final approval): [**`index.ipynb`**](https://github.com/trexfeathers/tephi/blob/0c29e3cc688eedc7a260c6d69a47d27b528c3223/index.ipynb)
3. We previously offered API for limiting plots via pressure (called 'anchor'), but internally this became a blocker to Tephi's further development, so we have replaced it with the simpler `xylim` argument. Unfortunately this is the other thing that got missed during the version 0.4 upgrade, so the documentation is misleading - will be fixed soon in #253.

`xylim` defines the bottom-left and top-right corners of the plot, in terms of two `(temperature, entropy)` pairs. It takes a bit of trial and error to line up with your desired pressures, but the line below delivers what you need:

```python
tpg = tephi.TephiAxes(xylim=[(0, 0), (-8, 51)])
```

In future we hope to bring back pressure-limiting of plots: #228 

Incidentally: we were unaware of the [notebook.community](https://notebook.community/) project. It's interesting and exciting to find another shop window that our software might appear in - increases the importance of Jupyter notebooks in our GitHub repos, in terms of providing them, and of maintaining them. Thanks for the heads up 😊

**Reply by hilaryweller0** [reply]
Awesome, thanks. I will share your code with our students before our field trip
